### ISSUE TITLE
MetaData Access

### ISSUE THREAD

**Issue by rstoneback** [pysat/pysat#274]
**Describe the bug**
The statement, ivm.meta.data = ivm.meta[keep, :] 
fails, when keep is a list of strings.

**To Reproduce**
1. See above.

**Expected behavior**
I should be able to down select from the full meta data object, a partial list of variables.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: MacOS
 - Version 10.14.5
 
**Additional context**

**Comment by jklenzing** [comment]
Trying to duplicate.
```
import pysat
test = pysat.Instrument('pysat', 'testing')
test.load(date=pysat.datetime(2009, 1, 1))
test.meta['uts']
```
yields
```
units                        s
long_name       Universal Time
desc                          
label                      uts
axis                       uts
scale                   linear
notes                         
value_min                  NaN
value_max                  NaN
fill                       NaN
custom                   False
Bin_Location               NaN
children                  None
Name: uts, dtype: object
```
Running `test.meta[['uts', 'slt']]` instead yields
```
AttributeError                            Traceback (most recent call last)
<ipython-input-9-182a36607d6f> in <module>
----> 1 test.meta[['uts', 'mlt']]

~/code/pysat/pysat/_meta.py in __getitem__(self, key)
    539         else:
    540             # ensure variable is present somewhere
--> 541             if key in self:
    542                 # get case preserved string for variable name
    543                 new_key = self.var_case_name(key)

~/code/pysat/pysat/_meta.py in __contains__(self, other)
    335         """case insensitive check for variable name"""
    336 
--> 337         if other.lower() in [i.lower() for i in self.keys()]:
    338             return True
    339         if other.lower() in [i.lower() for i in self.keys_nD()]:

AttributeError: 'list' object has no attribute 'lower'
```
Does this match what you get?

**Comment by rstoneback** [comment]
Yep.

**Comment by asher-pembroke** [comment]
Do we also want to support higher-order metdata slicing like `test.meta[:,:,:]` or `test.meta[:,keep, 'units']`? If so, should I return a mult-indexed dataframe?

**Comment by rstoneback** [comment]
No, thanks for the option though. MetaData is in a semi-weird state with the higher-order metadata required to deal with pandas DataFrames within Frames. I think the long term solution is to transition those data sets to xarray.

**Comment by jklenzing** [comment]
Closed with merge of #311
